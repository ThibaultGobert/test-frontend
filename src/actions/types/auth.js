export const LOGIN_START = 'LOGIN_START';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const LOGIN_SUCCES = 'LOGIN_SUCCESS';
export const LOGOUT = 'LOGOUT';
