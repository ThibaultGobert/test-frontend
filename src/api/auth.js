import axios from 'axios';
import baseUrl from './baseUrl';

class AuthApi {
  static login(credentials) {
    return axios.request({
      method: 'get',
      url: baseUrl + '/webresources/v1/authUser',
      timeout: 3000,
      headers: {
        'x-username': credentials.username,
        'x-password': credentials.password,
      }
    }).then(response => {
      return response.data;
    }).catch(error => {
      if(error.code == "ECONNABORTED") {
        throw "Timeout error";
      }

      let errorReturned = error.response.data;
      if (errorReturned.status == 500) {
        throw "Geen geldige username - wachtwoord combinatie";
      }

      throw error;
    });
  }
}

export default AuthApi;

// TODO: send pasword and username in a JSON body instead of using custom headers
