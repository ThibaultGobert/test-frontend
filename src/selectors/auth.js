import * as userRoles from '../constants/roles';

export const isStudent = state => {
  return (
    state.loggedIn.data.role === userRoles.STUDENT_ROLE ||
    state.loggedIn.data.role === userRoles.WORKSHOP_STUDENT_ROLE
  );
};
