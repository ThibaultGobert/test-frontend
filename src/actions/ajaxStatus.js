import * as types from './types';

export function beginAjaxCall(requestType, meta) {
  if (!requestType) {
    return { type: types.AJAX_CALL_BEGIN };
  }

  return { type: requestType, meta};
}

export function ajaxCallError(errorType, error) {
  return { type: errorType, error };
}
