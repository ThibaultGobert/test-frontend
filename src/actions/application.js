import { FINISH_BOOTSTRAP } from './types';

export function finishBootstrap() {
  return {
    type: FINISH_BOOTSTRAP
  };
}
