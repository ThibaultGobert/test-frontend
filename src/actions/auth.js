import { LOGIN_START, LOGIN_SUCCES, LOGIN_ERROR, LOGOUT } from './types';

export function loginStart() {
  return {
    type: LOGIN_START
  };
}

export function loginSuccess(response) {
  return {
    type: LOGIN_SUCCES,
    user: response
  };
}

export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error
  };
}

export function logOut() {
  return { type: LOGOUT };
}
