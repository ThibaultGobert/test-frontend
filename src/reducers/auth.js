import merge from 'lodash/merge';
import {
  LOGIN_START,
  LOGIN_SUCCES,
  LOGIN_ERROR,
  LOGOUT,
  FETCH_PROFILE_START,
  FETCH_PROFILE_SUCCESS,
  FETCH_PROFILE_ERROR,
} from '../actions/types';
import initialState from './initialState';

import { setUser, removeUser } from '../api/storage';

function authReducer(state = initialState.loggedIn, action) {
  switch (action.type) {
    case LOGIN_START:
      return { ...state, loading: true };

    case LOGIN_SUCCES: {
      setUser(action.user);
      return merge(state, { data: action.user, error: undefined, loading: false });
    }

    case LOGIN_ERROR:
      return { data: {}, error: action.error, loading: false };

    case FETCH_PROFILE_START:
      return merge({}, state, {
        personalInformation: { loading: true, error: null, hasError: false },
      });

    case FETCH_PROFILE_SUCCESS:
      return merge({}, state, {
        personalInformation: { loading: false, data: action.data },
      });

    case FETCH_PROFILE_ERROR:
      return merge({}, state, {
        personalInformation: { loading: false, error: action.error, hasError: true },
      });

    case LOGOUT: {
      removeUser();
      window.location = '/';
      return state;
    }

    default:
      return state;
  }
}

export default authReducer;
