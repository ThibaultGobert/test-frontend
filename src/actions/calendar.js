import * as types from './types';
import calendarApi from '../api/mockCalendar';
import {ajaxCallError, beginAjaxCall} from "./ajaxStatus";

export function loadCalendarSuccess(calendar) {
  return {type: types.LOAD_CALENDAR_SUCCESS, calendar };
}

export function loadCalendar() {
  return function(dispatch) {
    dispatch(beginAjaxCall());
    return calendarApi.getCalendar().then((calendar) => {
      dispatch(loadCalendarSuccess(calendar));
    }).catch(error => {
      dispatch(ajaxCallError());
      throw(error);
    });
  };
}
