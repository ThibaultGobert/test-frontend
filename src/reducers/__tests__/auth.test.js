import auth from '../auth';
import { loginSuccess } from '../../actions/auth';
import testReducer from '../../utils/test/testReducer';

const testAuthReducer = testReducer(auth);

it('should set user when login success', () => {
  let state = {
    data: {},
    loading: false,
    error: undefined
  };

  const action = loginSuccess({ name: 'Simon' });

  testAuthReducer(state, action);
});
