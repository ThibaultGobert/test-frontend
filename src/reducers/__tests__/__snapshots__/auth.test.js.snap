// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set user when login success 1`] = `
Object {
  "type": "LOGIN_SUCCESS",
  "user": Object {
    "name": "Simon",
  },
}
`;

exports[`should set user when login success 2`] = `
Object {
  "data": Object {
    "name": "Simon",
  },
  "error": undefined,
  "loading": false,
}
`;
